### PR TITLE
[SOAR-2241] - Added 'Quarantine Asset with Rapid7 Insight Agent from Slack' workflow

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Description: 

This PR adds the Quarantine Asset with Rapid7 Insight Agent from Slack workflow

## Validation: 
```
DCA-P50-3036:Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack jmcadams$ icon-validate .
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 29.058ms
```

## Screenshots: 
slack_flow1
workflow_loop1
workflow1

### CODE REVIEW

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Description
+
+This work flow allows for fast quarantine and unquarantine of asset that have an [Insight Agent](https://docs.rapid7.com/insight-agent/) installed on them.
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md]:
```suggestion
This workflow will quickly quarantine and unquarantine assets with the [Insight Agent](https://docs.rapid7.com/insight-agent/) installed on them via a Slack command and reports information back to Slack. The Insight Agent is available for deployment to any customer with InsightVM, InsightIDR, or InsightOps. This workflow identifies one or more assets in the form of IP addresses.
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+`@Rapid7 unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats with slack. 
+* Keep users happy by getting them back online quickly
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md]:
```suggestion
* Quarantine and unquarantine assets
* Keep users happy by getting them back online quickly
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats with slack.
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md]:
```suggestion
* Quickly respond to threats with slack
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+`@Rapid7 unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats with slack. 
+* Keep users happy by getting them back online quickly
+
+# Requirements
+
+* [Insight Agent](https://docs.rapid7.com/insight-agent/)
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:22]:
```suggestion
* [Slack](https://docs.rapid7.com/insightconnect/configure-slack-for-chatops/)
* [Insight Agent](https://docs.rapid7.com/insight-agent/)
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md (hunk 1 above)
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:3]:
```suggestion
This workflow allows for fast quarantine and unquarantine of an asset that has the [Insight Agent](https://docs.rapid7.com/insight-agent/) installed. 
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# Description
+
+This workflow allows for fast quarantine and unquarantine from Slack of an asset that has the [Insight Agent](https://docs.rapid7.com/insight-agent/) installed. 
+
+Sample Slack Trigger Commands:
+
+`@Rapid7 quarantine-endpoint 198.51.100.100"`
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:7]:
```suggestion
`@Rapid7 InsightConnect quarantine-endpoint 198.51.100.100"`
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+Sample Slack Trigger Commands:
+
+`@Rapid7 InsightConnect quarantine-endpoint 198.51.100.100"`
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:13]:
```suggestion
`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+# Description
+
+This workflow allows for fast quarantine and unquarantine from Slack of an asset that has the [Insight Agent](https://docs.rapid7.com/insight-agent/) installed. 
+
+Sample Slack Trigger Commands:
+
+`@Rapid7 InsightConnect quarantine-endpoint 198.51.100.100"`
+
+`@Rapid7 unquarantine-endpoint 198.51.100.100"`
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:9]:
```suggestion
`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100"`
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats with slack.
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:17]:
```suggestion
* Quickly respond to threats from Slack. 
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats from Slack.
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:17]:
```suggestion
* Quickly respond to threats by quarantining endpoints from Slack. 
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100"`
+
+For batches of assets, more than one IP can be specified
+
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats by quarantining endpoints from Slack. 
+* Keep users happy by getting them back online quickly
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:18]:
```suggestion
* Reduce the time it takes to contain an endpoint
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+# Key Features
+
+* Quickly respond to threats by quarantining endpoints from Slack. 
+* Reduce the time it takes to contain an endpoint
+
+# Requirements
+
+* [Insight Agent](https://docs.rapid7.com/insight-agent/)
+* [Insight Platform](https://docs.rapid7.com/insight/managing-platform-api-keys/)
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:23]:
```suggestion
* [Insight Platform API Key](https://docs.rapid7.com/insight/managing-platform-api-keys/)
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+|Plugin|Version|Count|
+|----|----|--------|
+|Rapid7 Insight Agent|1.0.0|3|
+
+## Troubleshooting
+
+_There is no troubleshooting information at this time_
+
+# Version History
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:61]: This is a different plugin's version history

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Activate the workflow in order to trigger with a Slack command.
+
+### Usage
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:33]: We don't need this section, it's duplicating the notes from the overview. Remove Usage section and add missing directions:

```
Once the workflow has been imported, Update the first step with the channel name to suit your Slack environment! by editing the input with the preset text of `change_me` to match the channel to monitor.
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/workflow.spec.yaml [hunk 14]
```diff
@@ -0,0 +1,23 @@
+extension: workflow
+products: ["insightconnect"]
+name: Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack
+title: "Quarantine Asset with Rapid7 Insight Agent from Slack"
+description: "This workflow uses a Slack command to quarantine or unquarantine an Insight Agent."
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/workflow.spec.yaml:5]:
```suggestion
description: "This workflow uses a Slack command to quarantine or unquarantine an endpoint monitored by the Insight Agent."
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+* [Insight Agent](https://docs.rapid7.com/insight-agent/)
+* [Insight Platform](https://docs.rapid7.com/insight/managing-platform-api-keys/)
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Activate the workflow in order to trigger with a Slack command.
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:31]:
```suggestion
Once the workflow has been imported, update the first step with the channel name to suit your Slack environment by editing the input with the preset text of change_me to match the channel to monitor.

After configuring the Slack steps, activate the workflow in order to trigger it with a Slack command.
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+* [Insight Platform API Key](https://docs.rapid7.com/insight/managing-platform-api-keys/)
+* [Slack](https://docs.rapid7.com/insightconnect/configure-slack-for-chatops/)
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported, Update the first step with the channel name to suit your Slack environment by editing the input with the preset text of `change_me` to match the channel to monitor.
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:31]:
```suggestion
Once the workflow has been imported, **update the first step with the channel name to suit your Slack environment** by editing the input with the preset text of `change_me` to match the channel to monitor.
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/workflow.spec.yaml [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+title: "Quarantine Asset with Rapid7 Insight Agent from Slack"
+description: "This workflow uses a Slack command to quarantine or unquarantine an endpoint monitored by the Insight Agent."
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response]
+  keywords: [quarantine, insight, agent, idr, ivm]
+  features: []
```
Comment (jschipp-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/workflow.spec.yaml:13]:
```suggestion
  features: [insightagent]
```

File: workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+`@Rapid7 InsightConnect unquarantine-endpoint 198.51.100.100 198.51.100.101 198.51.100.102"`
+
+# Key Features
+
+* Quickly respond to threats by quarantining endpoints from Slack
+* Reduce the time it takes to contain an endpoint
+
+# Requirements
+
+* [Insight Platform API Key](https://docs.rapid7.com/insight/managing-platform-api-keys/)
```
Comment (chogan-r7) [inline workflows/Quarantine_Asset_with_Rapid7_Insight_Agent_from_Slack/help.md:22]: Maybe we put this line in the requirements? Up to you 
```suggestion
* [Insight Platform API Key](https://docs.rapid7.com/insight/managing-platform-api-keys/) The Insight Agent is available for deployment to any customer with InsightVM, InsightIDR, or InsightOps.
```

**Pull request merged**